### DISCUSSION TITLE
Reactive data with Server Actions

### DISCUSSION THREAD

**Discussion by ixdi** [vercel/next.js#60769] [Help]
### Summary

Hello,

We are using Next.js 14.0.4 with App router. Do you know the correct way to update a reactive data for a component (the UI needs 'use client') from a database (we are using MongoDB Change Streams)?

I'm new to React and Next and I'm not sure if I just need a Server Action to update a data variable to pass as props to the Client Component.

We would like to use zustand as state managment in the browser, so maybe it's better to use socket.io, Server Side Events (SSE), or maybe another solution?

Thanks!

**Comment by icyJoseph** [comment]
Do you mean? https://react.dev/reference/react-dom/hooks/useFormState?

**Reply by ixdi** [reply]
No, we cannot use hooks, because our UI does not support server-side rendering. Also the data does not come from a form.
